### DISCUSSION TITLE
How do we want to provide our Flatpak apps?

### DISCUSSION THREAD

**Discussion by meisenzahl** [elementary/os#651] [Ideas]
Since https://github.com/elementary/appcenter/pull/1943, we have been using AppCenter to install Operating System Updates as Offline Updates. As we install the following flatpak apps via deb packaging, some problems arise:

- [elementary/browser](https://github.com/elementary/browser)
- [elementary/calculator](https://github.com/elementary/calculator)
- [elementary/camera](https://github.com/elementary/camera)
- [elementary/capnet-assist](https://github.com/elementary/capnet-assist)
- [elementary/evince](https://github.com/elementary/evince)
- [elementary/feedback](https://github.com/elementary/feedback)
- [elementary/font-viewer](https://github.com/elementary/font-viewer)
- [elementary/iconbrowser](https://github.com/elementary/iconbrowser)
- [elementary/music](https://github.com/elementary/music)
- [elementary/screenshot](https://github.com/elementary/screenshot)
- [elementary/videos](https://github.com/elementary/videos)

Due to the fact that now only via AppCenter the packages are downloaded and then installed after a reboot without an existing internet connection, the installation of the packages that try to install Flatpak Apps fails.

Currently, the installation of these packages fails because we are adding our Flatpak repository https://flatpak.elementary.io/repo.flatpakrepo if not already available. @tintou has suggested that we centralize this via a single package so that it is only run once when creating the OS image.

The installation of a single Flatpak app is done as follows:

```sh
flatpak install --system --noninteractive appcenter io.elementary.music//stable || true
```
Since we use `|| true` for this, the installation will not abort even if an error occurs. However, the installation of a Flatpak app can never be successful during an offline update due to the lack of an internet connection.

We have so far chosen to install our Flatpak apps via deb so that we can also bundle them via meta packages like elementary-desktop. Also, the current approach has the advantage that we can port packages from native to Flatpak even during the lifetime of an OS release.

The goal of this discussion is to find a solution so that we can hopefully take advantage of as many benefits as possible of the current solution, but also be able to do offline updates and offline upgrades.

**Comment by Letterus** [comment]
Thanks for that writing! As I'm not really into this, just a stupid question: Would it be possible to provide a local mirror of the Flatpak repo that contains the apps that need to be installed during offline installation? And after installation the repo source would be switched to the remote flatpak repo in case of availability of internet connectivity?

**Comment by ryonakano** [comment]
Closing as resolved since we've switched from Debian packaging to flatpak-preinstall to manage Flatpak apps as in https://github.com/elementary/os/issues/783. Flatpak apps no longer updated as a part of the offline updates.
